plugins/package/src/test/kotlin/net/corda/cli/plugins/packaging/CreateCpbTest.kt

```diff
@@ -0,0 +1,165 @@
+package net.corda.cli.plugins.packaging
+
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.StandardOpenOption
+import java.util.UUID
+import java.util.jar.Attributes
+import java.util.jar.JarEntry
+import java.util.jar.JarInputStream
+import java.util.jar.JarOutputStream
+import java.util.zip.ZipEntry
+import net.corda.cli.plugins.packaging.TestUtils.captureStdErr
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.io.TempDir
+import picocli.CommandLine
+
+class CreateCpbTest {
+
+    @TempDir
+    lateinit var tempDir: Path
+
+    private val app = CreateCpb()
+
+    companion object {
+        private val testKeyStore = Path.of(this::class.java.getResource("/signingkeys.pfx")?.toURI()
+            ?: error("signingkeys.pfx not found"))
+    }
+
+    private fun buildTestCpk(jars: List<String>): Path {
+        val cpkName = Path.of(tempDir.toString(), "${UUID.randomUUID()}.cpk")
+        JarOutputStream(Files.newOutputStream(cpkName, StandardOpenOption.CREATE_NEW)).use { jarOs ->
+            jars.forEach {
+                jarOs.putNextEntry(JarEntry(it))
+                jarOs.write("TEST CONTENT".toByteArray())
+            }
+            jarOs.putNextEntry(ZipEntry("META-INF/MANIFEST.MF"))
+            jarOs.write("TEST CONTENT".toByteArray())
+        }
+        return cpkName
+    }
+
+    @Test
+    fun `packs CPKs into CPB`() {
+        val cpk0 = buildTestCpk(
+            listOf(
+                "lib/cpk0-lib.jar",
+                "main-bundle0.jar"
+            )
+        )
+        val cpk1 =  buildTestCpk(
+            listOf(
+                "lib/cpk1-lib.jar",
+                "main-bundle1.jar"
+            )
+        )
+
+        val outcomeCpb = Path.of("$tempDir/cpb-test-outcome-${UUID.randomUUID()}.cpb")
```

Comment:
`tempDir` will be a new empty directory for every test. We could call the output file `cpb-test-outcome.cpb` because it will not see the files from other tests and other tests will not see it.

> _This only works when `tempDir` is on a non-static variable, which is the case here_